controllers/httproute_controller.go

```diff
@@ -253,6 +253,30 @@ func (r *HTTPRouteReconciler) reconcileHTTPRouteResource(ctx context.Context, ht
 		r.eventRecorder.Event(httpRoute.K8sObject(), corev1.EventTypeWarning, k8s.HTTPRouteventReasonFailedAddFinalizer, fmt.Sprintf("Failed add finalizer due to %v", err))
 	}
 
+	backendRefIPFamiliesErr := r.getBackendRefsIpFamiliesError(ctx, httpRoute.Inner())
+
+	if backendRefIPFamiliesErr != nil {
+		httpRouteOld := httpRoute.Inner().DeepCopy()
+
+		setupRouteStatusParents(httpRoute.Inner())
+
+		updateRouteCondition(httpRoute, metav1.Condition{
+			Type:               string(gateway_api.RouteConditionAccepted),
+			Status:             metav1.ConditionFalse,
+			ObservedGeneration: httpRoute.Inner().Generation,
+			Reason:             string(gateway_api.RouteReasonUnsupportedValue),
+			Message:            fmt.Sprintf("Dual stack Service is not supported"),
+		})
+
+		if err := r.client.Status().Patch(ctx, httpRoute.Inner(), client.MergeFrom(httpRouteOld)); err != nil {
+			glog.Errorf("updateHTTPRouteStatus: Patch() received err %s", err)
+
+			return errors.Wrapf(err, "failed to update httproute status")
+		}
+
+		return backendRefIPFamiliesErr
```

Comment:
can you explain how two errors above interact with each other:
`if backendRefIPFamiliesErr != nil and r.client.Status().Patch != nil`

it seems if backend error happens we should not get to Patch call, but we do in fact, or I'm missing something